users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -106,13 +119,66 @@ def add_specaug_layer(
     # Update config with pre-generated parameters
     full_config["specaug_params"] = specaug_params
 
+
     network[name] = {
         "class": "eval",
         "from": from_list,
         "eval": f"self.network.get_config().typed_value('transform')(source(0, as_data=True), network=self.network, **{full_config})",
     }
+    return [name], [
+        sort_filters_by_center_freq,
+        get_frequency_response,
+        filter_based_masking,
+        _mask,
+        random_mask,
+        transform,
+    ]
+
+
+def filter_based_masking(x, batch_axis, axis, probability_distribution, max_number_masks_for_probability_specaug):
```

Comment:
Did you test this sufficiently well? When I run an example with this function, it does not look like the distribution of masks follows what is given in `probability_distribution`. See the notebook that I provide you offline.

Reply:
This bug is fixed. See commit [b2dbb28](https://github.com/rwth-i6/i6_experiments/pull/246/commits/b2dbb28875fb45a4ea2f47c94a4a58e349a8ff97)